crates/runtime/src/embeddings/udtf.rs

```diff
@@ -354,6 +364,29 @@ impl TableFunctionImpl for VectorSearchTableFunc {
     }
 }
 
+/// This is a stub implementation, so that we can nest UDTF function invocations
+impl ScalarUDFImpl for VectorSearchTableFunc {
```

Comment:
nit: You could use `datafusion_expr::create_udf` macro, but then `scalar_invocation_error` would only be on `invoke_with_args` and not `return_type`.